python_modules/libraries/dagstermill/dagstermill/manager.py

```diff
@@ -190,6 +197,7 @@ def reconstitute_job_context(
                     job_context.for_step(
                         cast(ExecutionStep, execution_plan.get_step_by_key(step_key))
                     ),
+                    known_state=known_state
```

Comment:
`known_state` needs to be passed to `job_context.for_step` (it's currently passed to `cast`)